JsonEnvelopeSerializer/EnvelopeSerializer.cs

```diff
@@ -119,18 +119,24 @@ private JObject GenerateJObjectTreeFromObjectProperties(object value)
 
                 string tagName = attr.PropertyName ?? prop.Name;
 
-                if (null != propertyValue)
+                // Handle property-level [JsonProperty(NullValueHandling = NullValueHandling.Ignore)] attribute when value is null
+                if (propertyValue == null && attr.NullValueHandling == NullValueHandling.Ignore)
                 {
-                    retval.Add(tagName, JToken.FromObject(propertyValue));
                     continue;
                 }
 
-                // Handle property-level [JsonProperty(NullValueHandling = NullValueHandling.Ignore)] attribute when value is null
-                if (attr.NullValueHandling == NullValueHandling.Ignore)
+                // Handle property-level [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)] attribute when value is the default
+                if (attr.DefaultValueHandling == DefaultValueHandling.Ignore && propertyValue.Equals(GetDefaultValue(prop.PropertyType)))
```

Comment:
Can we just do:
```suggestion
                if (attr.DefaultValueHandling == DefaultValueHandling.Ignore && propertyValue.Equals(default(prop.PropertyType)))
```
And remove the GetDefaultValue method?

Reply:
`default` works with a type name but not a type instance.